homeassistant/components/husqvarna_automower_ble/config_flow.py

```diff
@@ -60,62 +64,218 @@ async def async_step_bluetooth(
         )
 
-    async def async_step_user(
+        title = await self.probe_mower(device)
+        if title is None:
+            return self.async_abort(reason="cannot_connect")
+
+        try:
+            errors: dict[str, str] = {}
+
+            (channel_id, mower) = await self.connect_mower(device)
+
+            response_result = await mower.connect(device)
+            await mower.disconnect()
+
+            if response_result is not ResponseResult.OK:
+                LOGGER.debug("cannot connect, response: {response_result}")
+
+                if (
+                    response_result is ResponseResult.INVALID_PIN
+                    or response_result is ResponseResult.NOT_ALLOWED
+                ):
+                    errors["base"] = "invalid_auth"
+                else:
+                    errors["base"] = "cannot_connect"
+
+                if self.source == SOURCE_BLUETOOTH:
+                    return self.async_show_form(
+                        step_id="bluetooth_confirm",
+                        data_schema=vol.Schema(
+                            {
+                                vol.Required(CONF_PIN): int,
+                            },
+                        ),
+                        errors=errors,
+                    )
+
+                suggested_values = {}
+
+                if self.address:
+                    suggested_values[CONF_ADDRESS] = self.address
+                if self.pin:
+                    suggested_values[CONF_PIN] = self.pin
+
+                return self.async_show_form(
+                    step_id="user",
+                    data_schema=self.add_suggested_values_to_schema(
+                        vol.Schema(
+                            {
+                                vol.Required(CONF_ADDRESS): str,
+                                vol.Required(CONF_PIN): int,
+                            },
+                        ),
+                        suggested_values,
+                    ),
+                    errors=errors,
+                )
+        except (TimeoutError, BleakError):
+            return self.async_abort(reason="cannot_connect")
+
+        return self.async_create_entry(
+            title=title,
+            data={
+                CONF_ADDRESS: self.address,
+                CONF_CLIENT_ID: channel_id,
+                CONF_PIN: self.pin,
+            },
+        )
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauthentication upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
-        """Handle the initial step."""
-        if user_input is not None:
+        """Confirm reauthentication dialog."""
+        errors: dict[str, str] = {}
+
+        if user_input:
```

Comment:
On other occasions you use:

```suggestion
        if user_input is not None:
```

Reply:
Fixed